app/src/main/java/de/danoeh/antennapod/discovery/ItunesTopListLoader.java

```diff
@@ -23,24 +24,46 @@
 import java.util.Locale;
 import java.util.concurrent.TimeUnit;
 
+import static android.content.Context.MODE_PRIVATE;
+
 public class ItunesTopListLoader {
     private static final String TAG = "ITunesTopListLoader";
     private final Context context;
+    public static final String PREF_KEY_COUNTRY_CODE = "country_code";
+    public static final String PREFS = "CountryRegionPrefs";
+    public static final String DISCOVER_HIDE_FAKE_COUNTRY_CODE = "00";
+    public static final String COUNTRY_CODE_UNSET = "99";
 
     public ItunesTopListLoader(Context context) {
         this.context = context;
     }
 
-    public Single<List<PodcastSearchResult>> loadToplist(int limit) {
+    public Single<List<PodcastSearchResult>> loadToplist() {
+        String defaultCountry = Locale.getDefault().getCountry();
+        SharedPreferences prefs = context.getSharedPreferences(PREFS, MODE_PRIVATE);
+        String countryCode = prefs.getString(PREF_KEY_COUNTRY_CODE, COUNTRY_CODE_UNSET);
+        return this.loadToplist(countryCode, 25);
+    }
+
+    public Single<List<PodcastSearchResult>> loadToplist(String country, int limit) {
         return Single.create((SingleOnSubscribe<List<PodcastSearchResult>>) emitter -> {
-            String country = Locale.getDefault().getCountry();
             OkHttpClient client = AntennapodHttpClient.getHttpClient();
-            String feedString;
+            String feedString = "{}";
```

Comment:
Is this still needed? I think there is no code path where the parser could end up with an empty string. The method either returns or throws an exception. While it wouldn't affect the runtime behavior, I think the code would be easier to understand if this was just kept uninitialized.